datastore/http/http.go

```diff
@@ -0,0 +1,234 @@
+	Address string `json:"address"`
+
+	// Timeout is the number of seconds to wait for a response. Defaults to 10.
+	Timeout int `json:"timeout"`
+
+	// TLS is the TLS configuration for the http producer.
+	TLS *TLSCertificate `json:"tls,omitempty"`
+}
+
+const (
+	statusCodePreSendErr = "NOT_SENT"
+	bufferSize           = 128
+)
+
+var (
+	metricsRegistry Metrics
+	metricsOnce     sync.Once
+)
+
+// NewProducer sets up an HTTP producer
+func NewProducer(config *Config, produceDecodedRecords bool, metricsCollector metrics.MetricCollector, namespace string, logger *logrus.Logger) (telemetry.Producer, error) {
+	registerMetricsOnce(metricsCollector)
+
+	err := validateConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	transport := http.DefaultTransport
+	if config.TLS != nil {
+		transport, err = getTransport(config.TLS)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	producer := &Producer{
+		namespace:             namespace,
+		metricsCollector:      metricsCollector,
+		workerChannels:        make([]chan *telemetry.Record, config.WorkerCount),
+		address:               config.Address,
+		logger:                logger,
+		produceDecodedRecords: produceDecodedRecords,
+		httpClient: http.Client{
+			Timeout:   time.Duration(config.Timeout) * time.Second,
+			Transport: transport,
+		},
+	}
+
+	for i := 0; i < config.WorkerCount; i++ {
+		producer.workerChannels[i] = make(chan *telemetry.Record, bufferSize)
+		go producer.worker(producer.workerChannels[i])
+	}
+
+	logger.Infof("registered http producer for namespace: %s", namespace)
+	return producer, nil
+}
+
+// validateConfig validates configuration values and sets defaults if value not set
+func validateConfig(c *Config) error {
+	if c.WorkerCount < 0 {
+		return errors.New("invalid http worker count")
+	}
+	if c.WorkerCount == 0 {
+		c.WorkerCount = 5
+	}
+	if c.Timeout < 0 {
+		return errors.New("invalid http timeout")
+	}
+	if c.Timeout == 0 {
+		c.Timeout = 10
+	}
+	return nil
+}
+
+func getTransport(tlsConfig *TLSCertificate) (*http.Transport, error) {
+	clientTLSCert, err := tls.LoadX509KeyPair(tlsConfig.CertFile, tlsConfig.KeyFile)
+	if err != nil {
+		return nil, err
+	}
+	return &http.Transport{
+		TLSClientConfig: &tls.Config{
+			Certificates: []tls.Certificate{clientTLSCert},
+		},
+	}, nil
+}
+
+func (p *Producer) worker(records <-chan *telemetry.Record) {
+	for record := range records {
+		metricsRegistry.bufferSize.Sub(1, map[string]string{})
+		p.sendHTTP(record)
+	}
+}
+
+func (p *Producer) sendHTTP(record *telemetry.Record) {
+	url := fmt.Sprintf("%s?namespace=%s&type=%s", p.address, p.namespace, record.TxType)
+	payload := record.Payload()
+
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(payload))
+	if err != nil {
+		p.logError(fmt.Errorf("create_request_err %s", err.Error()))
+		return
+	}
+
+	contentType := protobufContentType
+	if p.produceDecodedRecords {
+		contentType = jsonContentType
+	}
+	req.Header.Set("Content-Type", contentType)
```

Comment:
Updated to set `Content-Type` based on json vs protobuf. I'm using `application/x-protobuf` when it's the proto, but there are not clear standards on this available. Open to suggestion on this.